packages/melange-recharts/melange-recharts.4.0.0/opam

```diff
@@ -0,0 +1,50 @@
+opam-version: "2.0"
+synopsis: "Melange bindings for recharts"
+description: "Melange bindings for recharts JavaScript library."
+maintainer: [
+  "Liubomyr Mykhalchenko <liubomyr.mykhalchenko@ahrefs.com>"
+]
+authors: [
+  "Liubomyr Mykhalchenko <liubomyr.mykhalchenko@ahrefs.com>"
+]
+license: "MIT"
+tags: ["melange" "org:ahrefs"]
+homepage: "https://github.com/ahrefs/melange-recharts/"
+bug-reports: "https://github.com/ahrefs/melange-recharts/issues"
+depends: [
+  "dune" {>= "3.8"}
+  "ocaml"
+  "melange" {>= "2.0.0"}
+  "reason" {>= "3.10.0"}
+  "reason-react"
+  "reason-react-ppx"
+  "ocaml-lsp-server" {with-test}
+  "odoc" {with-doc}
+]
+build: [
+  ["dune" "subst"] {dev}
+  [
+    "dune"
+    "build"
+    "-p"
+    name
+    "-j"
+    jobs
+    "@install"
+    "@runtest" {with-test}
+    "@doc" {with-doc}
+  ]
+]
+dev-repo: "git+https://github.com/ahrefs/melange-recharts.git"
+depexts: [
+  ["recharts"] {npm-version = "^2.1.12"}
```

Comment:
it's perhaps missing a dependency to `opam-check-npm-deps`? (if my understanding of including this depexts here is correct)

Reply:
A package can define depexts to npm packages, but not need to check integrity of npm packages itself.

This is one of those cases: because `melange-recharts` doesn't include a `package.json` nor perform any kind of action / test using npm packages, we can skip the dep on `opam-check-npm-deps.

Reply:
I still haven't grasped the whole picture, but if it's intentional, this PR looks good to me.

(I'm guessing this is the documented [case for binding libraries](https://github.com/ahrefs/opam-check-npm-deps#for-bindings-authors-defining-constraints), but I cannot see how this package is different to say melange-moment and melange-numeral..)

Reply:
the tl;dr is: if your package has a `package.json` file (for dev purposes), then it should add `opam-check-npm-deps` as dep with `with-dev-setup`. Otherwise, there is no need.

Re: the examples you mentioned:
- `melange-moment` has [a package.json](https://github.com/ahrefs/melange-moment/blob/main/package.json), so in that case it makes sense to have `opam-check-npm-deps` as a dependency, to make sure at dev time that we have the right set of npm packages installed.
- You are right about `melange-numeral`. In that case, there is no `package.json`, so there was no need to have `opam-check-npm-deps`, I guess I got confused or missed it 😕 I can remove it from the opam file of the released version if needed.

In the case of `melange-recharts`, as there is no `package.json`, I didn't include `check-npm-deps`.

Please let me know if this helps clarifying!

Reply:
Ended up creating https://github.com/ocaml/opam-repository/pull/24664 to update `melange-numeral`.

Reply:
Thank you very much for explaining! It's much clearer for me now! (and sorry for making you create an extra PR 😅)

Reply:
np! thanks for the feedback while we figure this out 😄